custom_model_runner/datarobot_drum/resource/components/Python/uwsgi_component/uwsgi_serving.py

```diff
@@ -142,35 +145,40 @@ def predict(self, url_params, form_params):
         with tempfile.NamedTemporaryFile() as f:
             filestorage.save(f)
             f.flush()
-            out_df = self._predictor.predict(f.name)
-
-        num_columns = len(out_df.columns)
-        # float32 is not JSON serializable, so cast to float, which is float64
-        out_df = out_df.astype("float")
-        if num_columns == 1:
-            # df.to_json() is much faster.
-            # But as it returns string, we have to assemble final json using strings.
-            df_json = out_df[REGRESSION_PRED_COLUMN].to_json(orient="records")
-            response_json = '{{"predictions":{df_json}}}'.format(df_json=df_json)
-        elif num_columns == 2:
-            # df.to_json() is much faster.
-            # But as it returns string, we have to assemble final json using strings.
-            df_json_str = out_df.to_json(orient="records")
-            response_json = '{{"predictions":{df_json}}}'.format(df_json=df_json_str)
+            out_data = self._predictor.predict(f.name)
+
+        if self._unstructured_mode:
+            response = out_data
         else:
-            ret_str = (
-                "Predictions dataframe has {} columns; "
-                "Expected: 1 - for regression, 2 - for binary classification.".format(num_columns)
-            )
-            response_json = {"message": "ERROR: " + ret_str}
-            response_status = HTTP_422_UNPROCESSABLE_ENTITY
+            num_columns = len(out_data.columns)
```

Comment:
Hey, is this all the same validation that happens in the prediction_server.py?

Reply:
It was not introduced in this PR but since we touch this code, perhaps make sense to DRY this up?